includes/sanitizers/class-amp-img-sanitizer.php

```diff
@@ -368,38 +368,71 @@ private function adjust_and_replace_node( $node ) {
 	 */
 	private function maybe_add_lightbox_attributes( $attributes, $node ) {
 		$parent_node = $node->parentNode;
-		if ( ! ( $parent_node instanceof DOMElement ) || 'figure' !== $parent_node->tagName ) {
+		if ( ! ( $parent_node instanceof DOMElement ) || ! ( $parent_node->parentNode instanceof DOMElement ) ) {
 			return $attributes;
 		}
 
-		// Account for blocks that include alignment.
-		// In that case, the structure changes from figure.wp-block-image > img
+		// This should be true for links to the actual media file, but not for links to the attachment page.
+		$is_node_wrapped_in_media_file_link = (
+			'a' === $parent_node->tagName
+			&&
+			( 'figure' === $parent_node->tagName || 'figure' === $parent_node->parentNode->tagName )
+			&&
+			attachment_url_to_postid( $parent_node->getAttribute( 'href' ) )
```

Comment:
From what I  recall, `attachment_url_to_postid()` is not super reliable. I think it would be better to look at the URL to see if it looks like a file instead. For example:

```html
$is_file_url = preg_match( '/\.\w+$/', wp_parse_url( $parent_node->getAttribute( 'href' ), PHP_URL_PATH ) );
```

Reply:
Good idea!

Reply:
It's applied in https://github.com/ampproject/amp-wp/pull/3460/commits/783a8d8e239657f3f2eb265fe6a43b9313a28c84